cpu/sam0_common/sam0-eth/eth-netdev.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2020 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam0_common
+ * @{
+ *
+ * @file
+ * @brief       Low-level Ethernet driver implementation
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+#include <string.h>
+
+#include "mutex.h"
+#include "luid.h"
+
+#include "iolist.h"
+#include "net/gnrc/netif/ethernet.h"
+#include "net/gnrc.h"
+#include "net/ethernet.h"
+#include "net/netdev/eth.h"
+#include "net/eui_provider.h"
+
+#include "periph/gpio.h"
+
+#include "sam0_eth_netdev.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+#include "log.h"
+
+/* Internal helpers */
+extern int sam0_eth_init(void);
+extern int sam0_eth_send(const struct iolist *iolist);
+extern int sam0_eth_receive_blocking(char *data, unsigned max_len);
+extern void sam0_eth_set_mac(const eui48_t *mac);
+extern void sam0_eth_get_mac(char *out);
+
+/* SAM0 CPUs only have one GMAC IP, so it is safe to
+statically defines one in this file */
+sam0_eth_netdev_t _sam0_eth_dev;
```

Comment:
```suggestion
static sam0_eth_netdev_t _sam0_eth_dev;
```
I didn't see it being used outside this file